### ISSUE TITLE
Use of GitHub verb in footers

### ISSUE THREAD

**Issue by javier-godoy** [FlowingCode/DevelopmentConventions#9]
In #6 we discussed how to reference other issues from within the commit *body*
> Leaving aside the discussion on which verb we are using for closing issues (i.e. whether bugs are fixed and features are resolved, or evertything is just "closed")

The questions are:

A) Which GitHub verbs will be used as *footer*, for the purpose of closing an issue:
- close
- closes
- closed
- fix
- fixes
- fixed
- resolve
- resolves
- resolved

B) How do such verb align with the commit *type*? 
- Use the same keyword, irrespective of the commit type:
```
fix(ariane): handle overflow in conversion to 16-bit register

See http://www-users.math.umn.edu/~arnold/disasters/ariane.html

Close #nn
```

```
refactor: apply code changes that neither fix a bug nor add a feature

Close #nn
```
- Use different keywords, depending on the commit type:
```
fix(ariane): handle overflow in conversion to 16-bit register

See http://www-users.math.umn.edu/~arnold/disasters/ariane.html

Fix #nn
```

```
refactor: apply code changes that neither fix a bug nor add a feature

Resolve #nn
```

**Comment by javier-godoy** [comment]
My 2 cents:

> A) Which GitHub verbs will be used as footer, for the purpose of closing an issue:

* The chosen keyword(s) must keep with the imperative present tense that is required for the *subject* line and suggested for the commit *body*: `close`, `fix`, `resolve`.

> B) How do such verb align with the commit type?

* Use the same keyword, irrespective of the commit type. (Because it's easier to be consistent if there is a single option, and the purpose of a convention on commit messages is writing consistent commit messages.) 

* It cannot be `fix` because there are commits that don't fix anything (e.g. add a new feature).

* I'm slightly biased towards **`close`** since it clearly expresses the intention of **closing the issue** after the commit is applied (which is the only  purpose of writing `Close #nn` instead of `#nn`). Furthermore, [the definition of resolve](https://www.merriam-webster.com/dictionary/resolve) doesn't plug well with certain commit types:
  * If applied, this commit will **close/resolve** incomplete documentation
  * If applied, this commit will **close/resolve** inconsistent indentation
  * If applied, this commit will **close/resolve** a new feature


### TL;DR `Close`

**Comment by ngonzalezpazFC** [comment]
> A) Which GitHub verbs will be used as footer, for the purpose of closing an issue:

While I'm thinking about what to write at the commit subject, I try to complete the sentence "this commit will...", this way I feel  it natural. Following this excercise, it feels natural for me to think "this commit **closes**...", but this is a personal taste.

> B) How do such verb align with the commit type?

I agree with what Javier wrote previously about this question because GitHub does not provide all the keyworks needed to fit all of the commit types defined in the specification, and I also think **close** expresses the intention of closing the issue.

**Comment by mlopezFC** [comment]
I agree, it will make things easier. Independently of what the commit does, it will close the issue. If you want to know if we're fixing something or resolving something, that information is already in the commit type.

**Comment by javier-godoy** [comment]
Changed:

> Footers contains additional information about the commit, such as a description of the breaking changes, the list of issues that the commit will close, and other metadata.
> `Close #` provides a link to a single issue that is closed by the commit.
